eparticlectr/eparticlectr.cpp

```diff
@@ -175,11 +54,39 @@ void eparticlectr::brainclmid( uint64_t stakeid ) {
     stake_it = staketable.erase(stake_it);
 }
 
+// Redeem IQ, with a specific stake specified
+[[eosio::action]]
+void eparticlectr::brainclmidex( uint64_t stakeid ) {
+
+    // Get the stakes
+    staketblex staketable(_self, _self.value);
+    auto stake_it = staketable.find(stakeid);
+    eosio::check( stake_it != staketable.end(), "Stake does not exist in database");
+
+    // See if the stake is complete
+    eosio::check( eosio::current_time_point().sec_since_epoch() > stake_it->completion_time, "Staking period not over yet");
+
+    // Transfer back the IQ
+    asset iqAssetPack = asset(int64_t(stake_it->amount * IQ_PRECISION_MULTIPLIER), IQSYMBOL);
+    std::string memo = std::string("return stake #") + std::to_string(stake_it->id);
+    action(
+        permission_level{ _self, name("active") },
+        TOKEN_CONTRACT, name("transfrextra"),
+        std::make_tuple(_self, stake_it->user, iqAssetPack, memo, stake_it->proxied_for, std::string("return stake #") + std::to_string(stake_it->id), stake_it->extra_note )
+    ).send();
+
+    // Delete the stake.
+    // Note that the erase() function increments the iterator, then gives it back after the erase is done
+    stake_it = staketable.erase(stake_it);
+}
+
 
 // Place a vote using the IPFS hash
 // Users have to trigger this action through the everipediaiq::epartvote action
```

Comment:
is this because of keep tracking in everipedia DB of voting? otherwise it could also be an internal call

Reply:
yes